src/manager/YaraManager.js

```diff
@@ -0,0 +1,124 @@
+import yara from 'yara';
+import { promisify } from 'util';
+
+export default class YaraManager {
+  constructor() {
+    this.resources = [];
+  }
+
+  /*
+   * This Method scans a list of raw js artifacts using
+   * a set of Yara rules. Returns the Js artifacts that match
+   * otherwise, no malware was detected.
+   *
+   * If an error occurs in processing, values are pushed to the RO
+   * in order to be handled correctly by our api.
+   *
+   * When pushing errors to the RO data buffer:
+   *      error: boolean (whether or not an error has occurred in processing)
+   *      matchedRule: only valid if (isMalicious == true)
+   *      isMalicious: boolean, null if unknown
+   */
+  async setupResourceScan(jsArtifacts, ticket) {
+    console.log('Scanning JS...');
+    const ticketId = ticket.getID();
+    const responseFile = 'yara_response.txt';
+    const initialize = promisify(yara.initialize);
+
+    try {
+      // link to yara binary
+      await initialize();
+
+      const rules = [{ filename: 'src/resources/rules.yara' }];
+      const scanner = yara.createScanner();
```

Comment:
To what extent should we create promises? Is it mainly a timing thing? If we know a function may take a second it would be wise to promisify it to have execution wait for it?

Reply:
Honestly, you shouldn't have to mess with `Promise` too much. It's useful sometimes, because of things like `Promise.resolve` to create a promise that instantly resolves with a certain value for early-exit cases, or in this codebase I put in `Promise.all` in a few places to do work in parallel instead of how `async` pauses the current context.

You'll need to shell out to your own `Promise` if you're dealing with an instance of a class. That's why some devs prefer things like [bluebird](http://bluebirdjs.com/docs/features.html) to wrap old APIs, but I explained some reasons why not to include it above.